File: src/main/scala/com/sageserpent/americium/java/TestIntegrationContext.java
package com.sageserpent.americium.java;

public interface TestIntegrationContext<Case> {
    Case caze();

    CaseFailureReporting caseFailureReporting();

    InlinedCaseFiltration inlinedCaseFiltration();

    boolean isPartOfShrinkage();
}

File: src/main/java/com/sageserpent/americium/java/TestIntegrationContext.scala
package com.sageserpent.americium.java

trait TestIntegrationContext[+Case] {
  def caze: Case
  def caseFailureReporting: CaseFailureReporting
  def inlinedCaseFiltration: InlinedCaseFiltration
  def isPartOfShrinkage: Boolean
  def testCaseRecording: TestCaseRecording
}
